spec.html

```diff
@@ -27824,225 +27824,121 @@ <h1>Function Properties of the Math Object</h1>
+            1. If _n_ is *-0*, return *+0*.
+            1. If _n_ is *-&infin;*, return *+&infin;*.
+            1. If _n_ < 0, return -_n_.
+            1. Return _n_.
+        </emu-alg>
       </emu-clause>
 
       <emu-clause id="sec-math.acos">
         <h1>Math.acos ( _x_ )</h1>
         <p>Returns an implementation-approximated value representing the result of the arc cosine of _x_. The result is expressed in radians and ranges from *+0* to +&pi;.</p>
-        <ul>
-          <li>
-            If _x_ is *NaN*, the result is *NaN*.
-          </li>
-          <li>
-            If _x_ is greater than 1, the result is *NaN*.
-          </li>
-          <li>
-            If _x_ is less than -1, the result is *NaN*.
-          </li>
-          <li>
-            If _x_ is exactly 1, the result is *+0*.
-          </li>
-        </ul>
+        <emu-alg>
+            1. Let _n_ be ? ToNumber(_x_).
+            1. If _n_ is *NaN*, or _n_ &gt; 1, or _n_ &lt; -1, return *NaN*.
+            1. If _n_ is 1, return *+0*.
+            1. Return an implementation-approximated value representing the result of the arc cosine of _n_.
+        </emu-alg>
       </emu-clause>
 
       <emu-clause id="sec-math.acosh">
         <h1>Math.acosh ( _x_ )</h1>
         <p>Returns an implementation-approximated value representing the result of the inverse hyperbolic cosine of _x_.</p>
-        <ul>
-          <li>
-            If _x_ is *NaN*, the result is *NaN*.
-          </li>
-          <li>
-            If x is less than 1, the result is *NaN*.
-          </li>
-          <li>
-            If x is 1, the result is *+0*.
-          </li>
-          <li>
-            If _x_ is *+&infin;*, the result is *+&infin;*.
-          </li>
-        </ul>
+        <emu-alg>
+          1. Let _n_ be ? ToNumber(_x_).
+          1. If _n_ is *NaN*, or _n_ &lt; 1, return *NaN*.
+          1. If _n_ is 1, return *+0*.
+          1. If _n_ is *+&infin;*, return *+&infin;*.
+          1. Return an implementation-approximated value representing the result of the inverse hyperbolic cosine of _n_.
+        </emu-alg>
       </emu-clause>
 
       <emu-clause id="sec-math.asin">
         <h1>Math.asin ( _x_ )</h1>
         <p>Returns an implementation-approximated value representing the result of the arc sine of _x_. The result is expressed in radians and ranges from -&pi; / 2 to +&pi; / 2.</p>
-        <ul>
-          <li>
-            If _x_ is *NaN*, the result is *NaN*.
-          </li>
-          <li>
-            If _x_ is greater than 1, the result is *NaN*.
-          </li>
-          <li>
-            If _x_ is less than -1, the result is *NaN*.
-          </li>
-          <li>
-            If _x_ is *+0*, the result is *+0*.
-          </li>
-          <li>
-            If _x_ is *-0*, the result is *-0*.
-          </li>
-        </ul>
+        <emu-alg>
+          1. Let _n_ be ? ToNumber(_x_).
+          1. If _n_ is *NaN*, or _n_ &gt;1, or _n_ &lt; -1, return *NaN*.
```

Comment:
this also has two "or"s, as do a few others

Reply:
Neither; i'm saying that in any comma-separated list, only the final item should have a conjunction.

Reply:
These could also be collapsed together, although that may not be true in every case.
```suggestion
          1. If _n_ is *NaN* or abs(_n_) &gt; 1, return *NaN*.
```

Reply:
> Neither; i'm saying that in any comma-separated list, only the final item should have a conjunction.

It sounds like you're saying to use (e.g.):
`If _n_ is 1, 2, or 3, ...`
rather than:
`If _n_ is 1, or 2, or 3, ...`
I'm fine with that rule (and it pretty much agrees with usage in the spec).

However, it doesn't apply to the current case, which has the structure
`If A, or B, or C, ...`
where each of A/B/C is a separate 'condition'. Currently, for such cases, the spec is somewhat split on whether to use any commas, but it's quite consistent that all the `or`s should appear.

(It's less clear when the conjunction is `and`:
`If A and B and C, ...`
is about as common as
`If A, B, and C, ...`
)

Reply:
I wouldn't expect any differentiation between values or conditions; I'd expect what I'm used to in English sentences - that a comma-separated clause has only a final conjunction (and an Oxford comma, ofc).

Reply:
Well, then your expectation is at odds with current usage in the spec.

One downside of that style (in English and in the spec) is that you have to read most of the condition before you know whether the elided conjunction is "and" or "or". Which is not a big problem if the condition is short, but the spec has some long ones.

Reply:
In this PR, I've used the following format but I can adjust as needed:

```
1. If condition1, then ...
1. If condition1 or condition2, then ...
1. If condition1, condition2, or condition3, then ...
```